### PR TITLE
Functional components refactor

### PR DESCRIPTION
**Pull request by c3c1l1a**
The aim of this pull request is to refactor the code from class components to functional components.

#### What was accomplished
- The [Calculator.js](https://github.com/c3c1l1a/Math-Magician/blob/bde4a4eb22ca6b29322e690d6960393638b82314/src/components/Calculator.js) is now a functional component. This was achieved by using the the [useState hook](https://github.com/c3c1l1a/Math-Magician/blob/bde4a4eb22ca6b29322e690d6960393638b82314/src/components/Calculator.js#L10)

## Type of change
- New feature

## Checklist

-    [Linting tests](https://github.com/microverseinc/linters-config) pass locally with my changes
-    Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
-    Documentation is done in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,108 +1,56 @@
...
-        <DisplayScreen key="displayScreen" name="displayScreen" info={this.state} />
-        {KeyButtons}
-      </div>
-    );
-  }
+function Calculator() {
+  const [state, setState] = useState({
+    operand1: 0,
+    operand2: null,
+    operation: null,
```
Comment (codepantha) [inline src/components/Calculator.js:13]: - [ ] Please use ES6 arrow function syntax for your functional component 🙏

File: src/App.js [hunk 2]
```diff
@@ -2,12 +2,10 @@
...
-class App extends React.Component {
-  render() {
-    return (
-      <Calculator />
-    );
-  }
+function App() {
+  return (
+    <Calculator />
+  );
```
Comment (codepantha) [inline src/App.js:8]: - [ ] Please use ES6 arrow function syntax for your functional component 🙏

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUESTED CHANGES ♻️ 

## Hi @c3c1l1a 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Linters passing
✔️ App working as expected

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 3]
```diff
@@ -2,10 +2,8 @@
...
 import Calculator from './components/Calculator';
 
-function App() {
-  return (
-    <Calculator />
-  );
-}
+const App = () => (
+  <Calculator />
+);
```
Comment (mateo951) [inline src/App.js:7]: Addressed ✔

**Review by mateo951 — APPROVED** [review]
### Hi @c3c1l1a, 

Well done making the changes requested by the previous reviewer 🤓
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔥 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**